<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN"
"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en">
<head>
<!-- 2021-09-08 Wed 17:04 -->
<meta http-equiv="Content-Type" content="text/html;charset=utf-8" />
<meta name="viewport" content="width=device-width, initial-scale=1" />
<title>Site Visit - Data Dev</title>
<meta name="generator" content="Org mode" />
<meta name="author" content="John Brestelli" />
<style type="text/css">
 <!--/*--><![CDATA[/*><!--*/
  .title  { text-align: center;
             margin-bottom: .2em; }
  .subtitle { text-align: center;
              font-size: medium;
              font-weight: bold;
              margin-top:0; }
  .todo   { font-family: monospace; color: red; }
  .done   { font-family: monospace; color: green; }
  .priority { font-family: monospace; color: orange; }
  .tag    { background-color: #eee; font-family: monospace;
            padding: 2px; font-size: 80%; font-weight: normal; }
  .timestamp { color: #bebebe; }
  .timestamp-kwd { color: #5f9ea0; }
  .org-right  { margin-left: auto; margin-right: 0px;  text-align: right; }
  .org-left   { margin-left: 0px;  margin-right: auto; text-align: left; }
  .org-center { margin-left: auto; margin-right: auto; text-align: center; }
  .underline { text-decoration: underline; }
  #postamble p, #preamble p { font-size: 90%; margin: .2em; }
  p.verse { margin-left: 3%; }
  pre {
    border: 1px solid #ccc;
    box-shadow: 3px 3px 3px #eee;
    padding: 8pt;
    font-family: monospace;
    overflow: auto;
    margin: 1.2em;
  }
  pre.src {
    position: relative;
    overflow: auto;
    padding-top: 1.2em;
  }
  pre.src:before {
    display: none;
    position: absolute;
    background-color: white;
    top: -10px;
    right: 10px;
    padding: 3px;
    border: 1px solid black;
  }
  pre.src:hover:before { display: inline; margin-top: 14px;}
  /* Languages per Org manual */
  pre.src-asymptote:before { content: 'Asymptote'; }
  pre.src-awk:before { content: 'Awk'; }
  pre.src-C:before { content: 'C'; }
  /* pre.src-C++ doesn't work in CSS */
  pre.src-clojure:before { content: 'Clojure'; }
  pre.src-css:before { content: 'CSS'; }
  pre.src-D:before { content: 'D'; }
  pre.src-ditaa:before { content: 'ditaa'; }
  pre.src-dot:before { content: 'Graphviz'; }
  pre.src-calc:before { content: 'Emacs Calc'; }
  pre.src-emacs-lisp:before { content: 'Emacs Lisp'; }
  pre.src-fortran:before { content: 'Fortran'; }
  pre.src-gnuplot:before { content: 'gnuplot'; }
  pre.src-haskell:before { content: 'Haskell'; }
  pre.src-hledger:before { content: 'hledger'; }
  pre.src-java:before { content: 'Java'; }
  pre.src-js:before { content: 'Javascript'; }
  pre.src-latex:before { content: 'LaTeX'; }
  pre.src-ledger:before { content: 'Ledger'; }
  pre.src-lisp:before { content: 'Lisp'; }
  pre.src-lilypond:before { content: 'Lilypond'; }
  pre.src-lua:before { content: 'Lua'; }
  pre.src-matlab:before { content: 'MATLAB'; }
  pre.src-mscgen:before { content: 'Mscgen'; }
  pre.src-ocaml:before { content: 'Objective Caml'; }
  pre.src-octave:before { content: 'Octave'; }
  pre.src-org:before { content: 'Org mode'; }
  pre.src-oz:before { content: 'OZ'; }
  pre.src-plantuml:before { content: 'Plantuml'; }
  pre.src-processing:before { content: 'Processing.js'; }
  pre.src-python:before { content: 'Python'; }
  pre.src-R:before { content: 'R'; }
  pre.src-ruby:before { content: 'Ruby'; }
  pre.src-sass:before { content: 'Sass'; }
  pre.src-scheme:before { content: 'Scheme'; }
  pre.src-screen:before { content: 'Gnu Screen'; }
  pre.src-sed:before { content: 'Sed'; }
  pre.src-sh:before { content: 'shell'; }
  pre.src-sql:before { content: 'SQL'; }
  pre.src-sqlite:before { content: 'SQLite'; }
  /* additional languages in org.el's org-babel-load-languages alist */
  pre.src-forth:before { content: 'Forth'; }
  pre.src-io:before { content: 'IO'; }
  pre.src-J:before { content: 'J'; }
  pre.src-makefile:before { content: 'Makefile'; }
  pre.src-maxima:before { content: 'Maxima'; }
  pre.src-perl:before { content: 'Perl'; }
  pre.src-picolisp:before { content: 'Pico Lisp'; }
  pre.src-scala:before { content: 'Scala'; }
  pre.src-shell:before { content: 'Shell Script'; }
  pre.src-ebnf2ps:before { content: 'ebfn2ps'; }
  /* additional language identifiers per "defun org-babel-execute"
       in ob-*.el */
  pre.src-cpp:before  { content: 'C++'; }
  pre.src-abc:before  { content: 'ABC'; }
  pre.src-coq:before  { content: 'Coq'; }
  pre.src-groovy:before  { content: 'Groovy'; }
  /* additional language identifiers from org-babel-shell-names in
     ob-shell.el: ob-shell is the only babel language using a lambda to put
     the execution function name together. */
  pre.src-bash:before  { content: 'bash'; }
  pre.src-csh:before  { content: 'csh'; }
  pre.src-ash:before  { content: 'ash'; }
  pre.src-dash:before  { content: 'dash'; }
  pre.src-ksh:before  { content: 'ksh'; }
  pre.src-mksh:before  { content: 'mksh'; }
  pre.src-posh:before  { content: 'posh'; }
  /* Additional Emacs modes also supported by the LaTeX listings package */
  pre.src-ada:before { content: 'Ada'; }
  pre.src-asm:before { content: 'Assembler'; }
  pre.src-caml:before { content: 'Caml'; }
  pre.src-delphi:before { content: 'Delphi'; }
  pre.src-html:before { content: 'HTML'; }
  pre.src-idl:before { content: 'IDL'; }
  pre.src-mercury:before { content: 'Mercury'; }
  pre.src-metapost:before { content: 'MetaPost'; }
  pre.src-modula-2:before { content: 'Modula-2'; }
  pre.src-pascal:before { content: 'Pascal'; }
  pre.src-ps:before { content: 'PostScript'; }
  pre.src-prolog:before { content: 'Prolog'; }
  pre.src-simula:before { content: 'Simula'; }
  pre.src-tcl:before { content: 'tcl'; }
  pre.src-tex:before { content: 'TeX'; }
  pre.src-plain-tex:before { content: 'Plain TeX'; }
  pre.src-verilog:before { content: 'Verilog'; }
  pre.src-vhdl:before { content: 'VHDL'; }
  pre.src-xml:before { content: 'XML'; }
  pre.src-nxml:before { content: 'XML'; }
  /* add a generic configuration mode; LaTeX export needs an additional
     (add-to-list 'org-latex-listings-langs '(conf " ")) in .emacs */
  pre.src-conf:before { content: 'Configuration File'; }

  table { border-collapse:collapse; }
  caption.t-above { caption-side: top; }
  caption.t-bottom { caption-side: bottom; }
  td, th { vertical-align:top;  }
  th.org-right  { text-align: center;  }
  th.org-left   { text-align: center;   }
  th.org-center { text-align: center; }
  td.org-right  { text-align: right;  }
  td.org-left   { text-align: left;   }
  td.org-center { text-align: center; }
  dt { font-weight: bold; }
  .footpara { display: inline; }
  .footdef  { margin-bottom: 1em; }
  .figure { padding: 1em; }
  .figure p { text-align: center; }
  .equation-container {
    display: table;
    text-align: center;
    width: 100%;
  }
  .equation {
    vertical-align: middle;
  }
  .equation-label {
    display: table-cell;
    text-align: right;
    vertical-align: middle;
  }
  .inlinetask {
    padding: 10px;
    border: 2px solid gray;
    margin: 10px;
    background: #ffffcc;
  }
  #org-div-home-and-up
   { text-align: right; font-size: 70%; white-space: nowrap; }
  textarea { overflow-x: auto; }
  .linenr { font-size: smaller }
  .code-highlighted { background-color: #ffff00; }
  .org-info-js_info-navigation { border-style: none; }
  #org-info-js_console-label
    { font-size: 10px; font-weight: bold; white-space: nowrap; }
  .org-info-js_search-highlight
    { background-color: #ffff00; color: #000000; font-weight: bold; }
  .org-svg { width: 90%; }
  /*]]>*/-->
</style>
<link rel="stylesheet" type="text/css" href="https://fniessen.github.io/org-html-themes/src/readtheorg_theme/css/htmlize.css"/>
<link rel="stylesheet" type="text/css" href="https://fniessen.github.io/org-html-themes/src/readtheorg_theme/css/readtheorg.css"/>
<script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.3/jquery.min.js"></script>
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
<script type="text/javascript" src="https://fniessen.github.io/org-html-themes/src/lib/js/jquery.stickytableheaders.min.js"></script>
<script type="text/javascript" src="https://fniessen.github.io/org-html-themes/src/readtheorg_theme/js/readtheorg.js"></script>
<script type="text/javascript">
// @license magnet:?xt=urn:btih:e95b018ef3580986a04669f1b5879592219e2a7a&dn=public-domain.txt Public Domain
<!--/*--><![CDATA[/*><!--*/
     function CodeHighlightOn(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.add("code-highlighted");
         target.classList.add("code-highlighted");
       }
     }
     function CodeHighlightOff(elem, id)
     {
       var target = document.getElementById(id);
       if(null != target) {
         elem.classList.remove("code-highlighted");
         target.classList.remove("code-highlighted");
       }
     }
    /*]]>*///-->
// @license-end
</script>
</head>
<body>
<div id="content">
<h1 class="title">Site Visit - Data Dev</h1>
<div id="table-of-contents">
<h2>Table of Contents</h2>
<div id="text-table-of-contents">
<ul>
<li><a href="#org65a08c5">Slide 1: Intro</a></li>
<li><a href="#org65b1b5d">Slide 2: Example RNASeq</a>
<ul>
<li><a href="#org37d7062">Automated Presentation of Data</a>
<ul>
<li><a href="#org0a67a29">JBrowse</a></li>
<li><a href="#orgb0e5700">Searches</a></li>
<li><a href="#org8360376">Visualizations</a></li>
</ul>
</li>
<li><a href="#orgf75dc6a">Extended Functionality</a></li>
</ul>
</li>
<li><a href="#org90fdc6e">Slide 3: Example Comparative Genomics</a>
<ul>
<li><a href="#org18eb0e6">OrthoMCL Workflow</a>
<ul>
<li><a href="#orgd9a809b">Search for OrthoMCL Groups (Phyletic pattern)</a></li>
<li><a href="#orgd921c53">Transform Results from one organism to another on component sites</a></li>
<li><a href="#org2060dfa">Infer functional annotation</a></li>
</ul>
</li>
<li><a href="#org29ecd0f">Synteny</a>
<ul>
<li><a href="#org154ca78">Nextflow Workflow To process genome pairs in parallel (Syntenic Alignments)</a></li>
<li><a href="#org1098eaa">Synteny Views (JBrowse)</a></li>
</ul>
</li>
</ul>
</li>
<li><a href="#orgf41cd3f">Slide 4: Unified Database</a>
<ul>
<li><a href="#org83f783a">ebi2gus</a></li>
<li><a href="#orga2331b8">UniDB</a></li>
</ul>
</li>
</ul>
</div>
</div>
<div id="outline-container-org65a08c5" class="outline-2">
<h2 id="org65a08c5">Slide 1: Intro</h2>
<div class="outline-text-2" id="text-org65a08c5">
<ul class="org-ul">
<li>Introduce DD Group</li>
<li>Outline for remaining slides</li>
<li>all examples have been worked on in past year</li>
</ul>
</div>
</div>
<div id="outline-container-org65b1b5d" class="outline-2">
<h2 id="org65b1b5d">Slide 2: Example RNASeq</h2>
<div class="outline-text-2" id="text-org65b1b5d">
</div>
<div id="outline-container-org37d7062" class="outline-3">
<h3 id="org37d7062">Automated Presentation of Data</h3>
<div class="outline-text-3" id="text-org37d7062">
<p>
<b>The Data Development group uses the results of workflows to provide meaningful tools and visualizations to end users</b>
</p>
</div>
<div id="outline-container-org0a67a29" class="outline-4">
<h4 id="org0a67a29">JBrowse</h4>
<div class="outline-text-4" id="text-org0a67a29">
<ul class="org-ul">
<li>Templates so Browser Tracks can be added with minimal or zero configuration
<ul class="org-ul">
<li>Each Sample - Coverage</li>
<li>Each Dataset - Multi Coverage</li>
<li>Intron Junctions</li>
<li>Combined RNASeq</li>
</ul></li>
</ul>
</div>
</div>
<div id="outline-container-orgb0e5700" class="outline-4">
<h4 id="orgb0e5700">Searches</h4>
<div class="outline-text-4" id="text-orgb0e5700">
<ul class="org-ul">
<li>SQL Queries for Within Dataset Searches and templates so each Dataset gets its own searchs
<ul class="org-ul">
<li>Fold Difference (Find genes which are different between samples)</li>
<li>Expression Levels / Percentiles (Find genes which have high expression in one or more samples)</li>
</ul></li>
</ul>
</div>
</div>
<div id="outline-container-org8360376" class="outline-4">
<h4 id="org8360376">Visualizations</h4>
<div class="outline-text-4" id="text-org8360376">
<ul class="org-ul">
<li>R (plotly) visualizations of datasets
<ul class="org-ul">
<li>For each Gene, each dataset can be visualized individually or via a Transcriptomics Summary</li>
</ul></li>
</ul>
</div>
</div>
</div>
<div id="outline-container-orgf75dc6a" class="outline-3">
<h3 id="orgf75dc6a">Extended Functionality</h3>
<div class="outline-text-3" id="text-orgf75dc6a">
<p>
<b>The Data Development group writes workflow steps and front end automations as needed.  New workflows are using nextflow workflows and containerized software.</b>
</p>
<ul class="org-ul">
<li>Examples:
<ul class="org-ul">
<li>DESeq (pairwise comparisons)</li>
<li>JTK<sub>CYCLE</sub> (cyclical)</li>
<li>profile similarity (time series)</li>
<li>wgcna (co expression)</li>
</ul></li>
</ul>
</div>
</div>
</div>

<div id="outline-container-org90fdc6e" class="outline-2">
<h2 id="org90fdc6e">Slide 3: Example Comparative Genomics</h2>
<div class="outline-text-2" id="text-org90fdc6e">
</div>
<div id="outline-container-org18eb0e6" class="outline-3">
<h3 id="org18eb0e6">OrthoMCL Workflow</h3>
<div class="outline-text-3" id="text-org18eb0e6">
<p>
<b>The Data Development group (MarkH) redesigned the OrthoMCL workflow so each VEuPath organism is now processed and kept in sync with production releases.</b>
</p>
</div>
<div id="outline-container-orgd9a809b" class="outline-4">
<h4 id="orgd9a809b">Search for OrthoMCL Groups (Phyletic pattern)</h4>
<div class="outline-text-4" id="text-orgd9a809b">
<ul class="org-ul">
<li>Core Groups</li>
<li>Residual Groups</li>
</ul>
</div>
</div>
<div id="outline-container-orgd921c53" class="outline-4">
<h4 id="orgd921c53">Transform Results from one organism to another on component sites</h4>
<div class="outline-text-4" id="text-orgd921c53">
<ul class="org-ul">
<li>Create highly optimized tables to enable fast searches
<ul class="org-ul">
<li>a major improvment for scalability here in making the tuning table</li>
</ul></li>
</ul>
</div>
</div>
<div id="outline-container-org2060dfa" class="outline-4">
<h4 id="org2060dfa">Infer functional annotation</h4>
<div class="outline-text-4" id="text-org2060dfa">
<ul class="org-ul">
<li>Use groups to assign EC Numbers where this annotation was missing</li>
</ul>
</div>
</div>
</div>
<div id="outline-container-org29ecd0f" class="outline-3">
<h3 id="org29ecd0f">Synteny</h3>
<div class="outline-text-3" id="text-org29ecd0f">
<p>
<b>The Data Development group writes workflows and uses the results to provide meaningful tools and visualizations to end users</b>
</p>
</div>
<div id="outline-container-org154ca78" class="outline-4">
<h4 id="org154ca78">Nextflow Workflow To process genome pairs in parallel (Syntenic Alignments)</h4>
<div class="outline-text-4" id="text-org154ca78">
<ul class="org-ul">
<li>Schema Development and Loader were made to work with larger genomes and scale for the expected volume of genomes</li>
</ul>
</div>
</div>
<div id="outline-container-org1098eaa" class="outline-4">
<h4 id="org1098eaa">Synteny Views (JBrowse)</h4>
<div class="outline-text-4" id="text-org1098eaa">
<ul class="org-ul">
<li>We wrote a JBrowse Plugin to visualize syntenic spans and highlight ortholmcl groups (add picture)</li>
</ul>
</div>
</div>
</div>
</div>

<div id="outline-container-orgf41cd3f" class="outline-2">
<h2 id="orgf41cd3f">Slide 4: Unified Database</h2>
<div class="outline-text-2" id="text-orgf41cd3f">
</div>
<div id="outline-container-org83f783a" class="outline-3">
<h3 id="org83f783a">ebi2gus</h3>
<div class="outline-text-3" id="text-org83f783a">
<p>
<b>The Data Development group wrote the software that reads from many EBI Core MySQL Databases to populate a single "GUS" database instance</b>
</p>
<ul class="org-ul">
<li>Used for loading most genomes on VEuPath Sites</li>
<li>The container we made includes the Perl API and mysql server</li>
</ul>
</div>
</div>
<div id="outline-container-orga2331b8" class="outline-3">
<h3 id="orga2331b8">UniDB</h3>
<div class="outline-text-3" id="text-orga2331b8">
<p>
<b>The Same software is used to populate a single GUS instance from several GUS database instances</b>
</p>
<ul class="org-ul">
<li>Additional Data integrity Checks</li>
<li>not Oracle Specific.</li>
</ul>
</div>
</div>
</div>
</div>
<div id="postamble" class="status">
<p class="author">Author: John Brestelli</p>
<p class="date">Created: 2021-09-08 Wed 17:04</p>
<p class="validation"><a href="https://validator.w3.org/check?uri=referer">Validate</a></p>
</div>
</body>
</html>
